draft-ietf-tls-external-psk-importer.md

```diff
@@ -79,156 +80,151 @@ when, and only when, they appear in all capitals, as shown here.
-diversify a key based on some contextual information before use in a connection. In contrast to
-key exporters, wherein differentiation is done via an explicit label and context string,
-the key importer defined herein uses a label and set of hash algorithms to
-differentiate an external PSK into one or more PSKs for use.
+Key importers mirror the concept of key exporters in TLS in that they diversify a key
+based on some contextual information before use in a connection. In contrast to key exporters,
+wherein differentiation is done via an explicit label and context string, the key importer
+defined herein uses an optional context string along with a target protocol and KDF
+identifier to differentiate an external PSK into one or more PSKs for use.
 
 Imported keys do not require negotiation for use, as a client and server will not agree upon
 identities if not imported correctly. Thus, importers induce no protocol changes with
-the exception of expanding the set of PSK identities sent on the wire. Endpoints may incrementally
-deploy PSK importer support by offering non-imported keys for TLS versions prior to TLS 1.3.
-(Negotiation and use of imported PSKs requires both endpoints support the importer API described herein.)
+the exception of expanding the set of PSK identities sent on the wire. Endpoints may
+incrementally deploy PSK importer support by offering non-imported keys for TLS versions
+prior to TLS 1.3. (Negotiation and use of imported PSKs requires both endpoints support
+the importer API described herein.)
+
+Clients which import external keys TLS MUST NOT use these keys for any other purpose.
+Moreover, each external PSK MUST be associated with at most one hash function.
 
 ## Terminology {#terminology}
 
-- External PSK (EPSK): A PSK established or provisioned out-of-band, i.e., not from a TLS connection, which is
-a tuple of (Base Key, External Identity, KDF). The associated KDF (and hash function) may be undefined.
+- External PSK (EPSK): A PSK established or provisioned out-of-band, i.e., not from a TLS
+connection, which is a tuple of (Base Key, External Identity, Hash). The associated hash
+function may be undefined.
 - Base Key: The secret value of an EPSK.
 - External Identity: The identity of an EPSK.
 - Imported Identity: The identity of a PSK as sent on the wire.
+- Target protocol: The protocol for which a PSK is imported for use.
+- Target KDF: The KDF for which a PSK is imported for use.
 
 # Key Import
 
-A key importer takes as input an EPSK with external identity 'external_identity' and base key 'epsk',
-as defined in {{terminology}}, along with an optional label, and transforms
-it into a set of PSKs and imported identities for use in a connection based on supported HashAlgorithms.
-In particular, for each supported HashAlgorithm 'hash', the importer constructs an ImportedIdentity
-structure as follows:
+A key importer takes as input an EPSK with external identity `external_identity` and base key `epsk`,
```

Comment:
and a hash function.

Reply:
That's covered in the remainder of this sentence.